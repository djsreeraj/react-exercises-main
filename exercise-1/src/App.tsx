import { FC } from 'react'
import Time from './components/Time'

const App:FC  = () => {
    return (
        <div>
           <Time />
        </div>
    )
}

export default App
