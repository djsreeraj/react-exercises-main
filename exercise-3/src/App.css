@font-face {
  font-family: Roboto-Bold;
  src: url(https://fonts.gstatic.com/s/roboto/v27/KFOlCnqEu92Fr1MmWUlfBBc4AMP6lQ.woff2);
}

.countries {
  width: 100%;
  height: 100%;
  position: absolute;
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  font-family: Roboto-Bold;
  color: #37474f;
}

.countries__details,
.coutries__continent {
  display: flex;
  flex-direction: column;
  margin: 100px;
  background-color: whitesmoke;
  padding: 20px;
  border-radius: 5px;
}

h4 {
  font-size: 16px;
  margin-bottom: 20px;
}

input {
  margin-bottom: 50px;
  border-radius: 5px;
  padding: 10px;
  font-size: 1rem;
  outline: none;
  background-color: rgb(190, 237, 255);
  border: none;
  color: rgb(31, 103, 185);
}

ul {
  list-style: none;
}

li {
  padding: 10px;
}

.coutries__continent__results {
  height: 30vh;
  width: 30vh;
  overflow: auto;
}
