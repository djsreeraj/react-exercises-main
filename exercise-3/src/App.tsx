import { FC } from "react";
import Country from "./components/Country";
import Continent from "./components/Continent";
import "./App.css";

const App: FC = () => {
  return (
    <div className="countries">
      <Country />
      <Continent />
    </div>
  );
};

export default App;
