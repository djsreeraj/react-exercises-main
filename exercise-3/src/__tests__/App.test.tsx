import App from "../App";
import TestRenderer from "react-test-renderer";

it("renders correctly", () => {
  const component = TestRenderer.create(<App />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
