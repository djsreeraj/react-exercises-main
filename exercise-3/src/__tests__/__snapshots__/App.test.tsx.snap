// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="countries"
>
  <div
    className="countries__details"
  >
    <input
      onChange={[Function]}
      placeholder="Enter Country Code ( eg: IN )"
      type="text"
      value=""
    />
    <div
      className="coutries__continent__results"
    >
      <p>
         
      </p>
    </div>
  </div>
  <div
    className="coutries__continent"
  >
    <input
      onChange={[Function]}
      placeholder="Enter Continent Code ( eg: EU )"
      type="text"
      value=""
    />
    <div
      className="coutries__continent__results"
    >
      <p>
         
      </p>
    </div>
  </div>
</div>
`;
